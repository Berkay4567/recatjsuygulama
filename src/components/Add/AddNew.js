import React,{ useState} from 'react';
import { Link } from "react-router-dom";
const AddNew=()=>{
	let url = `https://5fc9346b2af77700165ae514.mockapi.io/simpsons`;
	
	//---Tanımlar----
	const[name,setName]=useState("");
	const[job,setJob]=useState("");
	const[about,setAbout]=useState("");
	const[avatar,setAvatar]=useState("");
	//---------------

    return(
		<div className="container">
			<div>
				<h1 className="text-center mt-5">Add a new Simpsons</h1>
				<form>
					<div className="form-group">
						<label>Name Surname:</label>
						<input
							type="name"
							className="form-control"
							placeholder="Full Name"
							onChange={e => setName(e.target.value)}
						/>
					</div>
					<div className="form-group">
						<label>Job Title:</label>
						<input
							type="job"
							className="form-control"
							placeholder="Enter job title"
							onChange={e => setJob(e.target.value)}
						/>
					</div>
					<div className="form-group">
						<label>About Him/Her:</label>
						<input
							type="about"
							className="form-control"
							placeholder="Enter about him/her"
							onChange={e => setAbout(e.target.value)}
						/>
					</div>
					<div className="form-group">
						<label>Image Link</label>
						<input
							type="img"
							className="form-control"
							placeholder="Enter image link"
							onChange={e => setAvatar(e.target.value)}
						/>
					</div>
                    <br/>
					<Link to={"/"}>
						<button
							type="button"
							className="btn btn-primary form-control"
							onClick={()=>ekle(name, job, about, avatar)}
						>
							Add Character
						</button>
					</Link>
					<Link className="mt-3 w-100 text-center" to="/" >
						or get back to contacts
					</Link>
				</form>
			</div>
		</div>

    )
	//------------------EKLEME KOMUTU-----------------------
	function ekle(name, job, about, avatar){
		try{
			return fetch(url,{
				method:"post",
				headers:{"Content-type": "application/json"},
				body: JSON.stringify({
					name:name,
					job:job,
					about:about,
					avatar:avatar
				})
			})
			.then(response => response.json())
			.then(data => {
				console.log("data",data)
	
			})
			
	
		}
		catch(err){
			console.log(err);
		}
	}
	//------------------EKLEME KOMUTU BİTİMİ------------------
}

export default AddNew