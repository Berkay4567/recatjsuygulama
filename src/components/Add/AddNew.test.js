import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router} from 'react-router-dom';
import AddNew from './AddNew';

it("renders when there are no items", ()=>{
    const tree=renderer.create(<Router>
        <AddNew />
      </Router>
  ).toJSON();
    expect(tree).toMatchSnapshot();
})