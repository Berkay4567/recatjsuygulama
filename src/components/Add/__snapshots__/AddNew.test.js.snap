// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are no iems 1`] = `
<div
  className="container"
>
  <div>
    <h1
      className="text-center mt-5"
    >
      Add a new Simpsons
    </h1>
    <form>
      <div
        className="form-group"
      >
        <label>
          Name Surname:
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Full Name"
          type="name"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Job Title:
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter job title"
          type="job"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          About Him/Her:
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter about him/her"
          type="about"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Image Link
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter image link"
          type="img"
        />
      </div>
      <br />
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="btn btn-primary form-control"
          onClick={[Function]}
          type="button"
        >
          Add Character
        </button>
      </a>
      <a
        className="mt-3 w-100 text-center"
        href="/"
        onClick={[Function]}
      >
        or get back to contacts
      </a>
    </form>
  </div>
</div>
`;

exports[`renders when there are no irems 1`] = `
<div
  className="container"
>
  <div>
    <h1
      className="text-center mt-5"
    >
      Add a new Simpsons
    </h1>
    <form>
      <div
        className="form-group"
      >
        <label>
          Name Surname:
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Full Name"
          type="name"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Job Title:
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter job title"
          type="job"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          About Him/Her:
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter about him/her"
          type="about"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Image Link
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter image link"
          type="img"
        />
      </div>
      <br />
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="btn btn-primary form-control"
          onClick={[Function]}
          type="button"
        >
          Add Character
        </button>
      </a>
      <a
        className="mt-3 w-100 text-center"
        href="/"
        onClick={[Function]}
      >
        or get back to contacts
      </a>
    </form>
  </div>
</div>
`;

exports[`renders when there are no items 1`] = `
<div
  className="container"
>
  <div>
    <h1
      className="text-center mt-5"
    >
      Add a new Simpsons
    </h1>
    <form>
      <div
        className="form-group"
      >
        <label>
          Name Surname:
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Full Name"
          type="name"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Job Title:
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter job title"
          type="job"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          About Him/Her:
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter about him/her"
          type="about"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Image Link
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter image link"
          type="img"
        />
      </div>
      <br />
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="btn btn-primary form-control"
          onClick={[Function]}
          type="button"
        >
          Add Character
        </button>
      </a>
      <a
        className="mt-3 w-100 text-center"
        href="/"
        onClick={[Function]}
      >
        or get back to contacts
      </a>
    </form>
  </div>
</div>
`;
